sdk/include/stdio.h

```diff
@@ -65,10 +66,19 @@ static inline int printf(const char *format, ...)
 #endif
 
 int __cheri_libcall snprintf(char *str, size_t size, const char *format, ...);
-int __cheri_libcall vsnprintf(const char *str,
+int __cheri_libcall vsnprintf(char *str,
                               size_t      size,
                               const char *format,
                               va_list     ap);
+
+static inline int sprintf(char *str, const char *format, ...)
+{
+	va_list args;
+	va_start(args, format);
+	int ret = vsnprintf(str, cheri_top_get(str) - cheri_address_get(str), format, args);
```

Comment:
I thought about creating a helper in `cheri-builtins.h` for top - address e.g. `cheri_remaining_get`, would you like that?